src/lib.rs

```diff
@@ -30,6 +31,45 @@ mod nodes;
 // TODO(jmmv): Make configurable via a flag and store inside SandboxFS.
 pub const TTL: Timespec = Timespec { sec: 60, nsec: 0 };
 
+/// An error indicating that a path has to be absolute but isn't.
+#[derive(Debug, Fail)]
+#[fail(display = "path {:?} is not absolute", path)]
+pub struct PathNotAbsoluteError {
+    /// The path that caused this error.
+    pub path: PathBuf,
+}
+
+/// Mapping describes how an individual path within the sandbox is connected to an external path
+/// in the underlying file system.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Mapping {
+    path: PathBuf,
+    underlying_path: PathBuf,
+    writable: bool,
+}
+
+impl Mapping {
+    /// Creates a new mapping from the individual components.
+    ///
+    /// `path` is the inside the sandbox's mount point where the `underlying_path` is exposed.
+    /// Both must be absolute paths.
+    pub fn new(path: &Path, underlying_path: &Path, writable: bool)
+        -> Result<Mapping, PathNotAbsoluteError> {
+        if !path.is_absolute() {
+            return Err(PathNotAbsoluteError { path: path.to_path_buf() });
+        }
+        if !underlying_path.is_absolute() {
+            return Err(PathNotAbsoluteError { path: underlying_path.to_path_buf() });
+        }
+
+        Ok(Mapping {
+            path: PathBuf::from(path),
```

Comment:
Why not to_path_buf like the above? Generally ::from is not used -- .into() is used instead.